python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -103,102 +198,81 @@ def register_launch_commands(self):
-                        "Software engine %s is not loaded in the current environment. "
-                        "Cannot launch %s" % (app_engine, app_path)
-                    )
-                    continue
-
-            # Download the thumbnail to use as the app's icon.
-            if app_icon:
-                sg_icon = shotgun_data.ShotgunDataRetriever.download_thumbnail(
-                    app_icon, self._tk_app
+        :returns: List of dictionaries containing required information to register
+                  a command with the current engine.
+        """
+        # List of command data to return
+        commands = []
+        if path:
+            # Register a command for each version for the path specified.
+            for version in (versions or [None]):
+                commands.append(self._register_command_data(
+                    display_name, icon, engine, path, args, version
+                ))
+        else:
+            try:
+                # Instantiate a SoftwareLauncher for the requested engine
+                # to see if it can determine a list of executable paths to
+                # register launch commands for.
+                launcher = sgtk.platform.create_engine_launcher(
+                    self._tk_app.sgtk, self._tk_app.context, engine
                 )
-                app_icon = sg_icon
-                self._tk_app.log_debug("App icon from ShotgunDataRetriever : %s" % app_icon)
-
-            # Parse the Software versions field to determine the specific list of
-            # versions to load. Assume the list of versions is stored as comma-separated
-            # in Shotgun.
-            app_versions = sw_entity["sg_versions"]
-            if app_versions:
-                for app_version in [v.strip() for v in app_versions.split(",")]:
-                    if not app_version :
-                        continue
-                    self._register_launch_command(
-                        app_menu_name, app_icon, app_engine, app_path, app_args, app_version
-                    )
-            else:
-                self._register_launch_command(
-                    app_menu_name, app_icon, app_engine, app_path, app_args
+                if launcher:
+                    # Get a list of SoftwareVersions for this engine and use that
+                    # data to construct a launch command to register.
+                    sw_versions = launcher.scan_software(versions, display_name, icon)
+                    for swv in sw_versions:
+                        commands.append(self._register_command_data(
+                            swv.display_name, swv.icon, engine, swv.path, args, swv.version
+                        ))
+            except Exception, e:
+                self._tk_app.log_warning(
+                    "Unable to determine path(s) for app [%s] from SoftwareLauncher "
+                    "for engine %s. Exception raised :\n%s" %  (display_name, engine, e)
```

Comment:
given that this will eventually be read by an artist, wonder if it's clearer if the message says something like


Cannot determine the path to tk-nuke: *Error message from scan_software*